assignment-2/src/view/SmartQuestionListByTag.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from "react";
+import questionModel from "../model/QuestionModel";
+import QuestionList from "./QuestionList";
+import questionListPresenter from "../presenter/QuestionListPresenter";
+
+const mapModelStateToComponentState = (modelState, props) => ({
+    questions: modelState.questions,
+    filter: props.match.params.filter
+});
+
+function filterByTag(questions, tag) {
```

Comment:
This filtering logic should be inside the model. In the worst case it should be inside the mapModelStateToComponentState.